src/Perspex.Base/Data/ValidationStatus.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) The Perspex Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Perspex.Data
+{
+    /// <summary>
+    /// Contains information on if the current object passed validation.
+    /// Subclasses of this class contain additional information depending on the method of validation checking. 
+    /// </summary>
+    public abstract class ValidationStatus
```

Comment:
The message would depend on the type of validation (that was the initial motivation for the abstract/derived class setup). Exception uses Exceptions, and INDEI uses an `IEnumerable`, so there's some incongruities. I can add another member and have them construct it separately. I was thinking about doing it in UI theming and design for each type (i.e. in Perspex.Themes.Default).